azurerm/internal/services/securitycenter/device_security_group_resource.go

```diff
@@ -0,0 +1,571 @@
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"target_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"allow_list_rule": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(security.RuleTypeConnectionToIPNotAllowed),
+								string(security.RuleTypeLocalUserNotAllowed),
+								string(security.RuleTypeProcessNotAllowed),
+							}, false),
+						},
+
+						"values": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+
+			"time_window_rule": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(security.RuleTypeActiveConnectionsNotInAllowedRange),
+								string(security.RuleTypeAmqpC2DMessagesNotInAllowedRange),
+								string(security.RuleTypeMqttC2DMessagesNotInAllowedRange),
+								string(security.RuleTypeHTTPC2DMessagesNotInAllowedRange),
+								string(security.RuleTypeAmqpC2DRejectedMessagesNotInAllowedRange),
+								string(security.RuleTypeMqttC2DRejectedMessagesNotInAllowedRange),
+								string(security.RuleTypeHTTPC2DRejectedMessagesNotInAllowedRange),
+								string(security.RuleTypeAmqpD2CMessagesNotInAllowedRange),
+								string(security.RuleTypeMqttD2CMessagesNotInAllowedRange),
+								string(security.RuleTypeHTTPD2CMessagesNotInAllowedRange),
+								string(security.RuleTypeDirectMethodInvokesNotInAllowedRange),
+								string(security.RuleTypeFailedLocalLoginsNotInAllowedRange),
+								string(security.RuleTypeFileUploadsNotInAllowedRange),
+								string(security.RuleTypeQueuePurgesNotInAllowedRange),
+								string(security.RuleTypeTwinUpdatesNotInAllowedRange),
+								string(security.RuleTypeUnauthorizedOperationsNotInAllowedRange),
+							}, false),
+						},
+
+						"max_threshold": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntAtLeast(0),
+						},
+
+						"min_threshold": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntAtLeast(0),
+						},
+
+						"time_window_size": {
```

Comment:
given the block already has time_window in it, should this just be window_size? and/or should this block be time_rule?

Reply:
personally I prefer "window_size", have updated